src/generic_app.ts

```diff
@@ -259,28 +390,89 @@ export class PolkadotGenericApp extends BaseApp {
-    return await this.signImpl(path, this.INS.SIGN, txBlob, txMetadata)
+    return await this.signImplEd25519(path, this.INS.SIGN, txBlob, txMetadata)
   }
   /**
+   * @deprecated Use signRawEcdsa or signRawEd25519 instead. This method will be removed in a future version.
    * Signs a raw transaction blob.
    * @param path - The BIP44 path.
    * @param txBlob - The transaction blob.
+   * @param scheme - The scheme to use for the signing. Default is ED25519.
    * @throws {ResponseError} If the response from the device indicates an error.
    * @returns The response containing the signature and status.
    */
-  async signRaw(path: BIP32Path, txBlob: TransactionBlob) {
-    return await this.signImpl(path, this.INS.SIGN_RAW, txBlob)
+  async signRaw(path: BIP32Path, txBlob: TransactionBlob, scheme = SCHEME.ED25519) {
+    if (scheme != SCHEME.ECDSA && scheme != SCHEME.ED25519) {
+      throw new ResponseError(LedgerError.ConditionsOfUseNotSatisfied, `Unexpected scheme ${scheme}. Needs to be ECDSA (2) or ED25519 (0)`)
+    }
+    if (scheme === SCHEME.ECDSA) {
+      return await this.signRawEcdsa(path, txBlob)
+    }
+    return await this.signRawEd25519(path, txBlob)
   }
 
   /**
+   * Signs a raw transaction blob using the ED25519 scheme.
+   * @param path - The BIP44 path.
+   * @param txBlob - The transaction blob.
+   * @throws {ResponseError} If the response from the device indicates an error.
+   * @returns The response containing the signature and status.
+   */
+  async signRawEd25519(path: BIP32Path, txBlob: TransactionBlob) {
+    return await this.signImplEd25519(path, this.INS.SIGN_RAW, txBlob)
+  }
+
+  /**
+   * Signs a raw transaction blob using the ECDSA scheme.
+   * @param path - The BIP44 path.
+   * @param txBlob - The transaction blob.
+   * @throws {ResponseError} If the response from the device indicates an error.
+   * @returns The response containing the signature and status.
+   */
+  async signRawEcdsa(path: BIP32Path, txBlob: TransactionBlob) {
+    return await this.signImplEcdsa(path, this.INS.SIGN_RAW, txBlob)
+  }
+
+  /**
+   * @deprecated Use signWithMetadataEd25519 or signWithMetadataEcdsa instead. This method will be removed in a future version.
    * [Expert-only Method] Signs a transaction blob with provided metadata (this could be used also with a migration app)
    * @param path - The BIP44 path.
    * @param txBlob - The transaction blob.
    * @param txMetadata - The transaction metadata.
    * @throws {ResponseError} If the response from the device indicates an error.
    * @returns The response containing the signature and status.
    */
-  async signWithMetadata(path: BIP32Path, txBlob: TransactionBlob, txMetadata: TransactionMetadataBlob) {
-    return await this.signImpl(path, this.INS.SIGN, txBlob, txMetadata)
+  async signWithMetadata(path: BIP32Path, txBlob: TransactionBlob, txMetadata: TransactionMetadataBlob, scheme: SCHEME) {
```

Comment:
Breaking change, not backwards compatible, adding `scheme: SCHEME` here as a param breaks everything still on 1.X.X

Reply:
It is not the case, as there is a default value provided. If the argument is not provided, as it is on v1.x.x, the scheme value will be the current one on v1.x.x, which is ED25519

Reply:
```suggestion
  async signWithMetadata(path: BIP32Path, txBlob: TransactionBlob, txMetadata: TransactionMetadataBlob, scheme: SCHEME) {
```

I don't see a default value here, it just says `scheme: SCHEME`.

Reply:
Thanks so much I appreciate it! Also thanks for getting ecdsa in :)

Reply:
Fix and more feature on its way #184